### PR TITLE
[Merged by Bors] - feat: the homology sequence of a distinguished triangle in the derived category

### PR DESCRIPTION
**Pull request by joelriou**
In this PR, we define `homologyFunctor C n : DerivedCategory C ⥤ C`, show they are homological functors and construct the long exact sequences associated to distinguished triangles in the derived category.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/Triangulated/HomologicalFunctor.lean [hunk 1]
```diff
@@ -140,6 +140,18 @@ noncomputable instance (priority := 100) [F.IsHomological] :
 instance (priority := 100) [F.IsHomological] : F.Additive :=
   F.additive_of_preserves_binary_products
 
+lemma isHomological_of_localization (L : C ⥤ D)
+    [L.CommShift ℤ] [L.IsTriangulated] [L.mapArrow.EssSurj ] (F : D ⥤ A)
```
Comment (erdOne) [inline Mathlib/CategoryTheory/Triangulated/HomologicalFunctor.lean:144]:
```suggestion
    [L.CommShift ℤ] [L.IsTriangulated] [L.mapArrow.EssSurj] (F : D ⥤ A)
```

File: Mathlib/Algebra/Homology/DerivedCategory/HomologySequence.lean [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+variable (T : Triangle (DerivedCategory C)) (hT : T ∈ distTriang _)
+  (n₀ n₁ : ℤ) (h : n₀ + 1 = n₁)
+
+/-- The connecting homomorphism on the homology sequence attached to a distinguished
+triangle in the derived category. -/
+noncomputable def δ : (homologyFunctor C n₀).obj T.obj₃ ⟶ (homologyFunctor C n₁).obj T.obj₁ :=
+  (homologyFunctor C 0).shiftMap T.mor₃ n₀ n₁ (by rw [add_comm 1, h])
+
+lemma comp_δ : (homologyFunctor C n₀).map T.mor₂ ≫ δ T n₀ n₁ h = 0 :=
```
Comment (erdOne) [inline Mathlib/Algebra/Homology/DerivedCategory/HomologySequence.lean:65]:
```suggestion
@[reassoc]
lemma comp_δ : (homologyFunctor C n₀).map T.mor₂ ≫ δ T n₀ n₁ h = 0 :=
```
Or
```suggestion
@[reassoc (attr := simp)]
lemma comp_δ : (homologyFunctor C n₀).map T.mor₂ ≫ δ T n₀ n₁ h = 0 :=
```

File: Mathlib/Algebra/Homology/DerivedCategory/HomologySequence.lean [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  (homologyFunctor C 0).comp_homologySequenceδ _ hT _ _ h
+
+lemma δ_comp : δ T n₀ n₁ h ≫ (homologyFunctor C n₁).map T.mor₁ = 0 :=
```
Comment (erdOne) [inline Mathlib/Algebra/Homology/DerivedCategory/HomologySequence.lean:68]:
```suggestion
@[reassoc]
lemma δ_comp : δ T n₀ n₁ h ≫ (homologyFunctor C n₁).map T.mor₁ = 0 :=
```
Or
```suggestion
@[reassoc (attr := simp)]
lemma δ_comp : δ T n₀ n₁ h ≫ (homologyFunctor C n₁).map T.mor₁ = 0 :=
```

File: Mathlib/Algebra/Homology/DerivedCategory/HomologySequence.lean (hunk 2 above)
Comment (joelriou) [reply Mathlib/Algebra/Homology/DerivedCategory/HomologySequence.lean:66]: Thanks! I have just added the reassoc/simp attribute.

**Comment by erdOne** [comment]
Thanks!
maintainer merge

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

**Pull request closed without merging**